packages/patternfly-4/react-table/src/components/Table/RowWrapper.js

```diff
@@ -77,7 +77,7 @@ class RowWrapper extends React.Component {
 }
 
 RowWrapper.propTypes = {
-  trRef: PropTypes.func,
+  trRef: PropTypes.object,
```

Comment:
not quite sure on this

Reply:
We need the function callbacks (`react-inline-edit-extension` uses that) and then chain the trRef callbacks. Please see `editableRowWrapper` and the usage of `trRef`. Is there something the functional approach is missing?

Reply:
For once, I think I agree w/ this [SO post](https://stackoverflow.com/questions/48007326/what-is-the-correct-proptype-for-a-ref-in-react). It depends on how you pass the ref in. Whether as a function like this:
```
ref={node => this.myRefToBePutInReduxGlobalStore = node}
```
or as a ref from `React.createRef()`:
```
<RowWrapper trRef={this.ref}
```
In this case since RowWrapper is extended, it can technically be either. Are you opposed to this for now:
```
PropTypes.oneOfType([
  PropTypes.func, 
  PropTypes.shape({ current: PropTypes.instanceOf(Element) })
])
```

Reply:
I would rather forbid of passing the ref object, because the ref might be needed by multiple extensions and the functional approach allows to get the ref for many extensions at once. On the other hand the object approach doesn't seem to support that. 

Example:

https://github.com/patternfly/patternfly-react/blob/96d033c4c5de26a8946885d44dcb99d8b7d90d72/packages/patternfly-4/react-inline-edit-extension/src/components/InlineEdit/editableRowWrapper.js#L219-L225

Reply:
This one is a toss up, i think i can refactor to callback refs in the future if desired. React doesn't seem to assume either way...

https://reactjs.org/docs/refs-and-the-dom.html#caveats-with-callback-refs

Reply:
I know, this just seems to be the easier way to use the ref across multiple components. 
Let me know if you can think of an alternative.

Reply:
fixed in https://github.com/patternfly/patternfly-react/pull/1250/commits/4a41d57b299535c7df2c1fc4b7a2ed0b062a1b83 and https://github.com/patternfly/patternfly-react/pull/1250/commits/f64e9be45e1799145c6a8f95d365c82e3651b164